Foundation/NSData.swift

```diff
@@ -75,20 +75,20 @@ public class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
     private var _base = _CFInfo(typeID: CFDataGetTypeID())
     private var _length: CFIndex = 0
     private var _capacity: CFIndex = 0
-    private var _deallocator: UnsafeMutablePointer<Void> = nil // for CF only
+    private var _deallocator: UnsafeMutablePointer<Void>? = nil // for CF only
     private var _deallocHandler: _NSDataDeallocator? = _NSDataDeallocator() // for Swift
-    private var _bytes: UnsafeMutablePointer<UInt8> = nil
+    private var _bytes: UnsafeMutablePointer<UInt8>? = nil
     
     internal var _cfObject: CFType {
         if self.dynamicType === NSData.self || self.dynamicType === NSMutableData.self {
             return unsafeBitCast(self, to: CFType.self)
         } else {
-            return CFDataCreate(kCFAllocatorSystemDefault, unsafeBitCast(self.bytes, to: UnsafePointer<UInt8>.self), self.length)
+            return CFDataCreate(kCFAllocatorSystemDefault, UnsafePointer<UInt8>(self.bytes), self.length)
         }
     }
     
     public override required convenience init() {
-        self.init(bytes: nil, length: 0, copy: false, deallocator: nil)
+        self.init(bytes: nil, length: 0, copy: true, deallocator: nil)
```

Comment:
This allocation is unfortunate, but necessary to preserve the property that NSData always has non-null `bytes`. There are several alternate solutions here, but this one requires the fewest changes elsewhere.